### PR TITLE
chore: Init multi actions

### PR DESCRIPTION
**Pull request by benjaminParisel**
* Adding packages folder to contains multiple actions
* Init some composite actions (notify-slack, list-branches)

### CODE REVIEW

File: .github/workflows/demo-list-branches.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  workflow_dispatch:
+
+jobs:
+  list-branches:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: List branches on repository
+        uses: bonitasoft/actions/packages/list-branches@main
+        with:
+          branche_regex: '^main$|test-flow'
```
Comment (rbioteau) [inline .github/workflows/demo-list-branches.yml:13]: 💡 Suggestion

Something like `selected_branches` and make it accept an array like this:
```yaml
 name: List branches on repository
        uses: bonitasoft/actions/packages/list-branches@main
        with:
          selected_branches: 
            - main
            - release-*
```

File: packages/list-branches/action.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    description: "Json object who contains each branch"
+    value: ${{ steps.extract_branches.outputs.branches }}
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Install jq and jo
```
Comment (rbioteau) [inline packages/list-branches/action.yml:19]: I think `jq` is already installed on the runner

File: packages/list-branches/action.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        sudo apt-get update
+        sudo apt-get install jq jo
+    - name: List branches
+      shell: bash
+      run: |
+        git remote set-branches origin '*'
+        git fetch
+        readarray branches <<< $(git branch --remote --format='%(refname:lstrip=3)' | tr '\n' ' ' | awk '{$1=$1};1')
+        json=$(jo -a $branches)
+        echo "##[set-output name=branches;]$(echo $json | jq -c 'map(select(test("${{inputs.branche_regex}}")))')"
```
Comment (rbioteau) [inline packages/list-branches/action.yml:31]: Regarding the suggestion above, try to transform the `selected_branches` into a regexp here

File: packages/notify-slack/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+name: Notify slack
+
+on: [pull_request]
+
+jobs:
+    notify-slack:
+        runs-on: ubuntu-latest
+        steps:
+          - name: Send message to Slack channel
+            if: ${{ failure() }}
```
Comment (rbioteau) [inline packages/notify-slack/README.md:25]: Can it be parametrized in the caller ?

File: packages/notify-slack/README.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+            uses: bonitasoft/actions/packages/notify-slack@main
+            with:
+              CHANNEL_ID: ${{ secrets.SLACK_UID_CHANNEL_ID }}
+              MESSAGE: ":x: Build on branch ${{github.ref}} failed"
+              SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
```
Comment (rbioteau) [inline packages/notify-slack/README.md:30]:
```suggestion
              SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
              TRIGGER_CONDITION: ${{ failure() }} // or ${{ always() }}...
```

**Review by rbioteau — COMMENTED** [review]
I have suggested some improvements.
I'll try to implement them when possible

File: packages/notify-slack/README.md (hunk 4 above)
Comment (benjaminParisel) [reply packages/notify-slack/README.md:25]: Yes it's, you can put what you want in the caller (i just put this example here to show a "real" use case")

File: packages/notify-slack/README.md (hunk 5 above)
Comment (benjaminParisel) [reply packages/notify-slack/README.md:30]: We don't need this trigger condition, you can already use `if` keyword in called step

File: packages/list-branches/action.yml (hunk 2 above)
Comment (benjaminParisel) [reply packages/list-branches/action.yml:19]: Yes it is ! I remove it

**Pull request merged**